bin/test_validator

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash -e
+
+ROOT_DIR=$(dirname $0)/..
+cd $ROOT_DIR
+
+if [[ $# != 1 ]]; then
+    echo Usage: $0 PROJECT_ID
+    false
+fi
+project_id=$1
+shift
+
+site_path=udmi_site_model
+device_id=AHU-1
+
+serial_no=validator-$RANDOM
+echo Using pubber with serial $serial_no
+
+PUBBER_OUT=pubber.out
+
+pids=`ps ax | fgrep pubber | fgrep java | awk '{print $1}'`
+if [[ -n $pids ]]; then
+    echo Killing pubber pids $pids
+    kill $pids
+fi
+
+rm -rf out/devices
+
+# Prepare auth key used by reflector
+cp udmi_site_model/devices/AHU-1/rsa_private.pkcs8 \
+   udmi_site_model/gcp_reflect_key.pkcs8
+
+bin/validator $site_path $project_id &
+vpid=$!
+sleep 10
+echo Started validator pid $vpid
+
+pubber/bin/build
+
+echo Writing pubber output to $PUBBER_OUT
+echo bin/pubber $site_path $project_id $device_id $serial_no
+bin/pubber $site_path $project_id $device_id $serial_no > $PUBBER_OUT 2>&1 &
+
+WAITING=10
+for i in `seq 1 $WAITING`; do
+    if fgrep "Connection complete" $PUBBER_OUT; then
+        break
+    fi
+    echo Waiting for pubber startup $((WAITING - i))...
+    sleep 2
+done
+
+if [[ $i == $WAITING ]]; then
+    echo pubber startup failed:
+    cat $PUBBER_OUT
+    false
+fi
+
+echo Waiting for system to run for a bit...
+sleep 30
+
```

Comment:
If this becomes more cumbersome, or we get unpredictable test results, I suggest using the shell's jobs command to focus our kill usage

For example:

jobs -x kill %pubber
jobs -x kill %validator

Although I'm not sure whether the shell would see either java or pubber/validator